### ISSUE TITLE
socket.io interface fails with multiple clients

### ISSUE THREAD

**Issue by MBeuttler** [IUNO-TDM/PaymentService#34]
When multiple clients subscribe to websocket events only one of them gets notified.

Happened when MarketplaceCore and MixerControl was running on the same machine, while only one PaymentService instance was runnung.

Current workarround: Run two instances of the paymentservice on two different ports.
Link the core and the mixer to different services.

**Comment by bockha** [comment]
@MBeuttler can you create a simple test, that checks if the payment service works properly?

**Comment by MBeuttler** [comment]
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Socket.IO Client</title>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.0.4/socket.io.js"></script>
    <script>

        var invoice = {
            totalAmount: 2000000,
            referenceId: 'Wurst',
            expiration: new Date(new Date().getTime() + (2 * 60 * 60 * 1000)).toISOString(),
            transfers: [{
                address: 'mi7RyP4PUc1TRaUCMo8yyY6FxGQ5C5S1yp',
                coin: 1000000
            }]
        };

        var socket = io('http://localhost:8080/invoices', {
            transports: ['websocket']
        });
        socket.on('connect', function(){
            console.log('Connected');


            var http = new XMLHttpRequest();
            var url = "http://localhost:8080/v1/invoices";
            http.open("POST", url, true);

            //Send the proper header information along with the request
            http.setRequestHeader("Content-type", "application/json");

            http.onreadystatechange = function() {//Call a function when the state changes.
                if(http.readyState === 4 && http.status === 201) {
                    console.log('Invoice Created:');
                    console.log(http.responseText);
                    const jsonData = JSON.parse(http.responseText);
                    socket.emit('room', jsonData['invoiceId']);
                }
            };
            http.send(JSON.stringify(invoice));

        });
        socket.on('StateChange', function(data){
            console.log('<StateChange>');
            console.log(data);
            console.log('<StateChange/>');
        });
        socket.on('disconnect', function(){
            console.log('Disconnected')
        });
    </script>

</head>
<body>
</body>
</html>
```

**Comment by bockha** [comment]
The socket.io events are passed as expected, I currently can not reproduce this issue. Refactoring is though necessary.

**Comment by bockha** [comment]
The problem has likely been caused by an empty wallet. The issue is likely fixed by dc8e5fe16053ed442a2476991c857eb5489e14d4 or 7468656eeb2dccb58975d7635524d324a2befb54
